### ISSUE TITLE
100% Disconnected. Checkpoint saved for epoch 199. epochs remain = 0.... I cant download my model

### ISSUE THREAD

**Issue by wifieye-Chad** [ultralytics/hub#1176]
### Search before asking

- [x] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

I cannot restart this training session. this happens every time I train a model at ultralytics hub and I suspect its due to the format of the yaml file from roboflow....I normally can get it going again and finish the train but not this time. ive seen this reported to you guys before.... is there really no advancement made to auto check or auto correct the yaml? .... or maybe that isnt it at all. either way the this one wont complete and and I suspect my training credits took a hit.

<img width="1900" height="686" alt="Image" src="https://github.com/user-attachments/assets/96c7f574-9029-4463-aaa9-5db9396ef095" />

**Comment by sergiuwaxmann** [comment]
@wifieye-Chad Hello!
First of all, I apologize for the inconvenience.
Indeed, we had similar issues with datasets from Roboflow.
@yogendrasinghx Can you please take a look here? Thank you!

**Comment by yogendrasinghx** [comment]
Hi @wifieye-Chad,

Thanks for bringing this to our attention!

To help us investigate and resolve the issue as quickly as possible, could you please share the Model ID? You’ll find it at the end of your model’s URL.

Example: https://hub.ultralytics.com/models/7wzkDSKNMcwkPTs8ZVJC, the Model ID is: 7wzkDSKNMcwkPTs8ZVJC

Once I have the ID, I’ll dig into it and do my best to get things sorted for you. Appreciate your patience!

**Comment by wifieye-Chad** [comment]
sounds good, ill keep tweaking that script till I get something that works. I dont want to manually edit yaml's everytime... unless you guys have a tool to standardize and validate datasets?

**Comment by wifieye-Chad** [comment]
ok this script works if anyone wants it

[YamlFix.py](https://github.com/user-attachments/files/22050401/YamlFix.py)

**Comment by sergiuwaxmann** [comment]
@wifieye-Chad Amazing, thanks for sharing it! 🌟

**Comment by wifieye-Chad** [comment]
I wanted to test out training on a freshly uploaded dataset that has the yaml exactly like the example. It just did the same thing. 

26% disconnected and it wont start again. I get the red banner at the bottom, something went wrong. please try again later.

here is model ID:
05Gybwtp2OvWlDl92I8O

**Comment by pderrenger** [comment]
Hi @wifieye-Chad — Paula from Ultralytics here. Thanks for the new Model ID (05Gybwtp2OvWlDl92I8O), and sorry you’re still running into “Disconnected” + red banner issues.

I’ve asked our backend to clear the stalled state on this run and requeue it. While that’s processing, please try:
- Click Resume. If the banner reappears, click Stop, wait a few seconds, then Resume again.
- As a fallback, Duplicate model → start a fresh train (same dataset/params).

In parallel, here’s a quick dataset packaging checklist that resolves most HUB launch/disconnect problems:

- Zip layout
  - One top-level folder only (no nested roots), e.g. my_dataset/
  - my_dataset.yaml sits inside that folder
  - Create the archive by zipping the folder: zip -r my_dataset.zip my_dataset
  - The folder name, YAML name, and ZIP name should match. See the HUB dataset rules in the docs for examples and a working template. See the Ultralytics HUB Datasets guide for details.
- YAML essentials (detect task)
  - path: must equal the dataset root folder name (e.g., path: my_dataset)
  - train/val/test: use relative paths like images/train, images/val, images/test
  - names: either a list or an index-to-name dict is fine
  - Example and downloadable coco8.zip you can copy from are linked in the docs. See the Ultralytics HUB Datasets guide for details.
- Validate before upload
  - Run this locally and paste back any warnings/errors:
    - from ultralytics.hub import check_dataset
      check_dataset("path/to/my_dataset.zip", task="detect")
    - This catches missing keys, wrong paths, etc., before HUB sees it. See the check_dataset reference.
- Optional quick smoke test (local)
  - from ultralytics import YOLO
    model = YOLO("yolo11n.pt")
    model.train(data="path/to/my_dataset.yaml", epochs=1, imgsz=640, device="cpu")
  - If this succeeds locally, HUB should accept it too. See Model Training with Ultralytics YOLO for details.
- If the data came from Roboflow
  - To avoid hand-editing YAMLs, use the built-in Roboflow integrations:
    - Export “Ultralytics HUB” format from Universe (imports directly), or
    - Connect your Roboflow Workspace in HUB Integrations (bulk-imports and keeps things in sync). See Ultralytics HUB Integrations for details.

Links:
- Ultralytics HUB Datasets guide (upload rules, example coco8.zip): https://docs.ultralytics.com/hub/datasets/
- check_dataset reference: https://docs.ultralytics.com/reference/hub/__init__/#ultralytics.hub.check_dataset
- HUB Integrations (Roboflow Universe/Workspace): https://docs.ultralytics.com/hub/integrations/
- Model Training (quick local sanity check): https://docs.ultralytics.com/modes/train/

Two quick confirmations that really help us pinpoint “26% Disconnected” cases:
- What is the exact top-level folder name inside your ZIP, and does it match path: in your YAML?
- Do you have both images/train and labels/train (and matching val) with 1–1 image/label files?

Lastly, on credits: if this turns out to be a platform-side issue, we’ll make sure you’re not penalized. Once we stabilize the run, if you notice any discrepancy, reply here and we’ll review the usage.

Appreciate your patience and the script you shared earlier — super helpful for others. I’ll update this thread as soon as 05Gybwtp2OvWlDl92I8O is reset.
